modal/_utils/pattern_matcher.py

```diff
@@ -151,44 +152,43 @@ def match(self, path: str) -> bool:
 class PatternMatcher:
     """Allows checking paths against a list of patterns."""
 
-    def __init__(self, patterns: list[str]) -> None:
+    def __init__(self, *pattern: str) -> None:
         """Initialize a new PatternMatcher instance.
 
         Args:
-            patterns (list): A list of pattern strings.
+            pattern (str): One or more pattern strings.
 
         Raises:
             ValueError: If an illegal exclusion pattern is provided.
         """
         self.patterns: list[Pattern] = []
         self.exclusions = False
-        for pattern in patterns:
-            pattern = pattern.strip()
-            if not pattern:
+        for p in list(pattern):
+            p = p.strip()
+            if not p:
                 continue
-            pattern = os.path.normpath(pattern)
+            p = os.path.normpath(p)
             new_pattern = Pattern()
-            if pattern[0] == "!":
-                if len(pattern) == 1:
+            if p[0] == "!":
+                if len(p) == 1:
                     raise ValueError('Illegal exclusion pattern: "!"')
                 new_pattern.exclusion = True
-                pattern = pattern[1:]
+                p = p[1:]
                 self.exclusions = True
             # In Python, we can proceed without explicit syntax checking
-            new_pattern.cleaned_pattern = pattern
-            new_pattern.dirs = pattern.split(os.path.sep)
+            new_pattern.cleaned_pattern = p
+            new_pattern.dirs = p.split(os.path.sep)
             self.patterns.append(new_pattern)
 
-    def matches(self, file_path: str) -> bool:
+    def matches(self, file_path: Path) -> bool:
```

Comment:
Perhaps name it `_matches` to indicate this is not part of the public interface of a pattern matcher in Modal.